### PR TITLE
Sync daily download counts

### PR DESCRIPTION
**Pull request by szakarias**

### CODE REVIEW

**Review by jonasfj — APPROVED** [review]

File: app/lib/service/download_counts/sync_download_counts.dart [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    dayCounts[version] = int.parse(count);
+  });
+
+  return dayCounts;
+}
+
+Future<bool> processDownloadCounts(
+    String downloadCountsFileName, DateTime date) async {
+  final downloadCountsBucket =
+      storageService.bucket(activeConfiguration.downloadCountsBucketName!);
```
[the remark this answers is not in the document]
Comment (jonasfj) [reply app/lib/service/download_counts/sync_download_counts.dart:49]: maybe what you have makes test easier.

File: app/lib/service/download_counts/sync_download_counts.dart [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  Uint8List bytes;
+  try {
+    bytes = await downloadCountsBucket.readAsBytes(downloadCountsFileName);
+  } on Exception {
+    _logger.info('Failed to read "$downloadCountsFileName".');
```
[the remark this answers is not in the document]
Comment (jonasfj) [reply app/lib/service/download_counts/sync_download_counts.dart:54]: or maybe not if it's the latest date...

File: app/lib/service/download_counts/sync_download_counts.dart [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      }
+    });
+  }));
+
+  if (failedLines) {
+    return false;
+  } else {
+    final allPackageNames = await packageBackend.allPackageNames().toSet();
+    final missingPackages = allPackageNames
+      ..removeWhere((p) => processedPackages.contains(p));
```
[the remark this answers is not in the document]
Comment (jonasfj) [reply app/lib/service/download_counts/sync_download_counts.dart:96]: or ignore this suggestion :D

File: app/lib/service/download_counts/sync_download_counts.dart [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+      return await pool.withResource(() async {
+        await downloadCountsBackend.updateDownloadCounts(package, {}, date);
+      });
+    }));
+    return true;
+  }
+}
+
+const numberOfSyncDays = 5;
+Future<void> syncDownloadCounts() async {
```
[the remark this answers is not in the document]
Comment (szakarias) [reply app/lib/service/download_counts/sync_download_counts.dart]: done

File: app/test/service/download_counts/download_counts_test.dart [hunk 5]
```diff
@@ -38,5 +46,268 @@ void main() {
...
+              'service',
+              'download_counts',
+              'fake_download_counts_data_faulty_line.jsonl'));
+      final logger = Logger('');
+      final messages = <String>[];
+      logger.onRecord.listen((event) {
+        messages.add(event.message);
+      });
+      final succeeded =
+          await processDownloadCounts(downloadCountsJsonFileNameJan6, nextDate);
```
[the remark this answers is not in the document]
Comment (szakarias) [reply app/test/service/download_counts/download_counts_test.dart]: done

**Review by jonasfj — APPROVED** [review]

**Pull request merged**